cpp/src/fil/fil.cu

```diff
@@ -57,14 +57,17 @@ void dense_node_decode(const dense_node_t* n, float* output, float* thresh,
 
 __host__ __device__ float sigmoid(float x) { return 1.0f / (1.0f + expf(-x)); }
 
-__global__ void transform_k(float* preds, size_t n, bool output_class,
-                            float threshold) {
+__global__ void transform_k(float* preds, size_t n, output_t output,
+                            float inv_num_trees, float threshold) {
   size_t i = threadIdx.x + size_t(blockIdx.x) * blockDim.x;
   if (i >= n) return;
-  float out = preds[i];
-  out = sigmoid(out);
-  if (output_class) out = out > threshold ? 1.0f : 0.0f;
-  preds[i] = out;
+  float pred = preds[i];
```

Comment:
I found the mix of pred and preds in one function a little confusing... maybe pred_array and pred?